### PR TITLE
Revert 'Make Y equivalent to y$.'

### PR DESCRIPTION
**Pull request by jb55**
from commit 5fc80917be117039b620ed5e4e79aa1f62b9b4bf #2818

We should strive to be as vim-like as possible, non-default bindings
on core keys such as Y should be discouraged imo.

Let me know what you guys think.

### CODE REVIEW

**Comment by zilongshanren** [comment]
@jb55 `yy` is for yanking the whole line and `Y` is equal to `y$` which is the same as `D` and `C` motions.  Both of them are mean to `operate to the end of this line`.

**Comment by jb55** [comment]
I understand that, yy and Y are synonyms in vim and that's what people expect. Custom bindings should be left to user configs.

**Comment by zilongshanren** [comment]
@jb55  The default is not evil.. And I think it's easy for you to override the key mapping.
And it seems most vimers like me already use the mapping for years.

**Comment by jb55** [comment]
It's evil when you change a key binding that has been around since 1976. I highly doubt "most vimers" are like you, I certainly am not.

**Comment by robbyoconnor** [comment]
Yeah this is one of those things that belongs in your `dotspacemacs/user-config` / `dotspacemacs/config` 

So :-1: -- ***PLEASE*** kill with fire.

**Comment by jb55** [comment]
Some other discussions on this issue:

kinda of for Y but decided on default for now: https://github.com/neovim/neovim/issues/416
had Y but dropped it: https://github.com/tpope/vim-sensible/issues/47#issuecomment-21985585
for Y: https://github.com/codemirror/CodeMirror/issues/2643

**Comment by zilongshanren** [comment]
@jb55  Thanks for the links. But Spacemacs is not Vim... So I keep my opinion and I already put the configs in my `dotspacemacs/user-config`. So this fix won't hurt me.

So @robbyoconnor  , there is no fire here :smile:

**Comment by kini** [comment]
FWIW, evil-mode itself does have a policy of keeping defaults as similar to vim as possible. So maybe what's more relevant is to say that spacemacs is not evil-mode, either.

**Comment by robbyoconnor** [comment]
Sorry, I like starting fires :)

**Comment by TheBB** [comment]
This is the first time in my 8 years of vim and vim-like editors I've seen anyone suggest that the default behavior of Y in vim was not a mistake.

**Comment by jb55** [comment]
I think everyone agrees that the original Y keybinding was a mistake. It's more about the decision of whether or not we care about vi/vim/evil compatibility. If not then we can close this. If we do close this, it will be a major point of friction for users coming from vi, vim and evil. Is it really worth it?

**Comment by jb55** [comment]
Some more data, I posted this to #vim on freenode and the vim subreddit: http://strawpoll.me/5534443

Quite controversial indeed, 50% split as of this comment.

**Comment by robbyoconnor** [comment]
Leave it open -- allow others to comment :)

**Comment by rakyi** [comment]
Spacemacs is pretty opinionated and already binds lots of other keys away from vim defaults (IMHO for the better), e.g. `SPC`, `\`, `,`, and I think mapping `y$` to `Y` should stay. Also too relevant: https://xkcd.com/1172/

**Comment by syl20bnr** [comment]
That's a hard decision since both views can be seen as consistent. One is consistent with Vi defaults and the other is consistent with the other Vi key bindings. 

So it looks like a perfect case for a dotspacemacs variable. Now what default value to set ?

I'm for a richer vocabulary and then having synonym is not very useful, and let's be honest that Y behavior consistency with other keys is the more relevant than historical consistency, at least in this case.

In summary let's add a variable `dotspacemacs-remap-Y-to-y$` which defaults to t.

**Comment by jb55** [comment]
OK I'll rework this PR sometime today

**Comment by robbyoconnor** [comment]
I actually used <kbd>Y</kbd> today....so I kinda like it.

**Comment by syl20bnr** [comment]
@jb55 why did you close this PR ?

**Comment by jb55** [comment]
Because we're not reverting it?

**Comment by syl20bnr** [comment]
OK I thought you wanted to add the variable. I'll do it tonight. I'm not really sure about the default value for this variable though.

**Comment by robbyoconnor** [comment]
I think the default should be the current behavior as we have it personally.

**Comment by yogsototh** [comment]
I just upgraded to 104 today and this is hurting my brain. Is the discussion closed? I'm really used to "Y" to copy current line with the newline. Do I have to change this behaviour in my .spacemacs?

**Comment by syl20bnr** [comment]
No I did not take the actions decided in the thread :-(
I will add the missing variable tonight and release a 0.104.1, sorry about this.

**Comment by yogsototh** [comment]
Great! Thanks!
